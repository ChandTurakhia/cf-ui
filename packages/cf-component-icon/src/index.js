import IconUnstyled from './Icon';
import IconTheme from './IconTheme';

import { applyTheme } from 'cf-style-container';

const Icon = applyTheme(IconUnstyled, IconTheme);

export { Icon, IconUnstyled, IconTheme };
