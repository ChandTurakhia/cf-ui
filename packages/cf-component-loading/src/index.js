import LoadingUnstyled from './Loading';
import LoadingTheme from './LoadingTheme';

import { applyTheme } from 'cf-style-container';

const Loading = applyTheme(LoadingUnstyled, LoadingTheme);

export { Loading, LoadingUnstyled, LoadingTheme };
