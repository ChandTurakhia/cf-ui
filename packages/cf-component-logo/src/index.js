import Logo from './Logo';
import logoSvg from '../assets/logo.svg';
import logoSvgMobile from '../assets/logo_mobile.svg';

export { Logo, logoSvg, logoSvgMobile };
