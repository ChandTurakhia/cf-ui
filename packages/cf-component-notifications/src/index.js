import NotificationUnstyled from './Notification';
import NotificationListUnstyled from './NotificationList';
import NotificationGlobalContainerUnstyled from './NotificationGlobalContainer';

import NotificationTheme from './NotificationTheme';
import NotificationListTheme from './NotificationListTheme';
import NotificationGlobalContainerTheme
  from './NotificationGlobalContainerTheme';

import { applyTheme } from 'cf-style-container';

const Notification = applyTheme(NotificationUnstyled, NotificationTheme);

const NotificationList = applyTheme(
  NotificationListUnstyled,
  NotificationListTheme
);

const NotificationGlobalContainer = applyTheme(
  NotificationGlobalContainerUnstyled,
  NotificationGlobalContainerTheme
);

export {
  Notification,
  NotificationList,
  NotificationGlobalContainer,
  NotificationUnstyled,
  NotificationListUnstyled,
  NotificationGlobalContainerUnstyled,
  NotificationTheme,
  NotificationListTheme,
  NotificationGlobalContainerTheme
};
