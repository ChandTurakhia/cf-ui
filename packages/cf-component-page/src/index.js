import PageUnstyled from './Page';
import PageHeaderUnstyled from './PageHeader';
import PageContentUnstyled from './PageContent';
import PageTheme from './PageTheme';
import Header from './Header';
import Logo from './Logo';
import NavList from './NavList';
import NavItem from './NavItem';
import logoSvg from './logo.svg';
import logoSvgMobile from './logo_mobile.svg';

import { applyTheme } from 'cf-style-container';

const Page = applyTheme(PageUnstyled, PageTheme);
const PageHeader = applyTheme(PageHeaderUnstyled, {});
const PageContent = applyTheme(PageContentUnstyled, {});

export {
  Header,
  Logo,
  logoSvg,
  logoSvgMobile,
  NavItem,
  NavList,
  Page,
  PageTheme,
  PageUnstyled,
  PageHeader,
  PageHeaderUnstyled,
  PageContent,
  PageContentUnstyled
};
