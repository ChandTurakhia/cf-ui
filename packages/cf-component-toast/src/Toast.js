import React from 'react';
import PropTypes from 'prop-types';
import { createComponent } from 'cf-style-container';

const getColors = (theme, type) => {
  switch (type) {
    case 'success':
      return {
        borderColor: theme.color.grass50,
        backgroundColor: theme.color.grass10
      };
    case 'error':
      return {
        borderColor: theme.color.apple50,
        backgroundColor: theme.color.apple10
      };
    case 'info':
      return {
        borderColor: theme.color.marine50,
        backgroundColor: theme.color.marine10
      };
  }
};

const ToastComponent = createComponent(({ theme, type, last }) => ({
  padding: theme.padding,
  marginBottom: last ? 0 : theme.marginBottom,
  borderWidth: theme.borderWidth,
  borderStyle: theme.borderStyle,
  ...getColors(theme, type)
}));

const Toast = props => <div><ToastComponent {...props} /></div>;

Toast.propTypes = {
  type: PropTypes.oneOf(['success', 'error', 'info']).isRequired,
  last: PropTypes.bool
};

export default Toast;
