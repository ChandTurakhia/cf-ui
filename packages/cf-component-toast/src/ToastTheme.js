export default baseTheme => ({
  padding: '0.5rem 0.8rem',
  marginBottom: '1rem',
  borderWidth: 1,
  borderStyle: 'solid'
});
