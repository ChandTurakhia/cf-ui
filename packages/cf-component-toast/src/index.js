import { applyTheme } from 'cf-style-container';

import ToastUnstyled from './Toast';
import ToastTheme from './ToastTheme';

const Toast = applyTheme(ToastUnstyled, ToastTheme);

export { Toast, ToastUnstyled, ToastTheme };
