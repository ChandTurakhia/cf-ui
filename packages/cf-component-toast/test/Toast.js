import React from 'react';
import { felaSnapshot } from 'cf-style-provider';
import { Toast } from '../../cf-component-toast/src/index';

test('should render success', () => {
  const snapshot = felaSnapshot(<Toast type="success">Success</Toast>);
  expect(snapshot.component).toMatchSnapshot();
  expect(snapshot.styles).toMatchSnapshot();
});

test('should render error', () => {
  const snapshot = felaSnapshot(<Toast type="error">Error</Toast>);
  expect(snapshot.component).toMatchSnapshot();
  expect(snapshot.styles).toMatchSnapshot();
});

test('should render info', () => {
  const snapshot = felaSnapshot(<Toast type="info">Info</Toast>);
  expect(snapshot.component).toMatchSnapshot();
  expect(snapshot.styles).toMatchSnapshot();
});
