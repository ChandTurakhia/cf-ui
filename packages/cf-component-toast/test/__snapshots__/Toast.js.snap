// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render error 1`] = `
<div>
  <div
    className="a b c d e f"
  >
    Error
  </div>
</div>
`;

exports[`should render error 2`] = `
"
.a {
  padding: 0.5rem 0.8rem
}

.b {
  margin-bottom: 1rem
}

.c {
  border-width: 1px
}

.d {
  border-style: solid
}

.e {
  border-color: #de9293
}

.f {
  background-color: #f8e9e9
}
"
`;

exports[`should render info 1`] = `
<div>
  <div
    className="a b c d e f"
  >
    Info
  </div>
</div>
`;

exports[`should render info 2`] = `
"
.a {
  padding: 0.5rem 0.8rem
}

.b {
  margin-bottom: 1rem
}

.c {
  border-width: 1px
}

.d {
  border-style: solid
}

.e {
  border-color: #97bddf
}

.f {
  background-color: #eaf2f8
}
"
`;

exports[`should render success 1`] = `
<div>
  <div
    className="a b c d e f"
  >
    Success
  </div>
</div>
`;

exports[`should render success 2`] = `
"
.a {
  padding: 0.5rem 0.8rem
}

.b {
  margin-bottom: 1rem
}

.c {
  border-width: 1px
}

.d {
  border-style: solid
}

.e {
  border-color: #cde49e
}

.f {
  background-color: #f5faeb
}
"
`;
